src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxTreeTests.cs

```diff
@@ -125,7 +134,7 @@ public void WithDiagnosticOptionsEmpty()
             Assert.NotSame(tree.DiagnosticOptions, newTree.DiagnosticOptions);
         }
 
-        [Fact]
+        [Fact, Obsolete]
```

Comment:
So this feels funky here to be marking the test as Obsolete -- sure, it's calling Obsolete members, but the behavior being tested here is unrelated to the reason that method is obsolete. I wouldn't want somebody to later come by and see this test and think it can just be deleted.

Should we at least have a convention something like:

1. Tests that are testing obsoleted APIs and are testing the behavior that was obsoleted can be marked [Obsolete]
2. Tests that are calling into APIs that are obsoleted for incidental reasons are either updated or #pragma warning disabled the specific points so the test itself doesn't appear "obsolete".

Reply:
I agree in general, but I'm not sure why this specific test shouldn't be Obsolete. If I remove the attribute I get this:

![image](https://user-images.githubusercontent.com/41759/183142010-e54f48c4-d2b2-49c5-bf3a-d28914aabe22.png)

The API being tested here are all obsolete.

Reply:
Changed to `[Fact, Obsolete("Testing obsolete API")]`